spec/slack/real_time/client_spec.rb

```diff
@@ -159,16 +159,31 @@
       end
 
       describe '#run_handlers' do
-        context 'when store has no event hooks' do
+        before do
+          @events = client.store.class.events.dup
+          client.store.class.events.clear
+        end
+
+        after do
+          client.store.class.events.merge!(@events)
+        end
+
+        context 'when config#async_handlers is :all', if: ENV['CONCURRENCY'] == 'async-websocket' do
           before do
-            @events = client.store.class.events.dup
-            client.store.class.events.clear
+            client.async_handlers = :all
+            allow(socket).to receive(:run_async).and_return(Async.run { nil })
```

Comment:
that first and last points are what i was talking about in my last comment, it doesn't return an `Async::Task` out of the box because the socket is mocked. It just gets `nil` back

Reply:
when i run without that explicit return i get this
```
Failures:

  1) Slack::RealTime::Client client with a full store #start! #run_handlers when config#async_handlers is :all returns an Async::Task
     Failure/Error: expect(client.send(:run_handlers, 'example', {})).to be_a Async::Task
       expected nil to be a kind of Async::Task
     # ./spec/slack/real_time/client_spec.rb:182:in `block (6 levels) in <top (required)>'
```

Reply:
This was expected. In this test the socket client (the part that is actually async) is mocked rather than using the real concurrency class.

```ruby
let(:ws) { double(Slack::RealTime::Concurrency::Mock::WebSocket, on: true) }
```

It therefore never invokes the real `Slack::RealTime::Concurrency::Async::Socket` and so you can't check the return of that.

The test here is good, it checks that 1) when `async_handlers=:all` we call `run_async` on the socket class, and 2) when `async_handlers=:none` we don't call it. If we want to check that `run_async` works as expected for the real `Slack::RealTime::Concurrency::Async::Socket` we take it out into a separate spec for that, which I added in https://github.com/milestruecar/slack-ruby-client/pull/1. I think we need to add tests there (in a future PR), which reminds me to do code coverage ;)